index.bs

```diff
@@ -3318,8 +3318,7 @@ Create a named {{MLOperand}} based on a descriptor, that can be used as an input
         The permissions and context validity have been checked by [[#api-mlgraphbuilder-constructor]] steps.
     </div>
     1. If |name| is empty, then [=exception/throw=] a {{TypeError}}.
-    1. [=Assert=]: If |descriptor|.{{MLOperandDescriptor/dimensions}} does not [=map/exist=], then |descriptor| defines a scalar input.
-    1. If |descriptor|.{{MLOperandDescriptor/dimensions}} [=map/exists=]:
+    1. If |descriptor|.{{MLOperandDescriptor/dimensions}}'s [=list/size=] is not 0:
```

Comment:
Can we just delete this extra conditional line? The 0-size case shouldn't need special handling here because "checking dimensions" should handle it correctly, but I should have deleted this line too in it:

`If dimensions’s [size](https://infra.spec.whatwg.org/#list-size) is 0, return false.`

The "checking dimensions" step is apparently only used by `input` and `constant(descriptor, bufferView)`.

(https://www.w3.org/TR/webnn/#ref-for-check-dimensions)

Then we could unindent line 3322 and just always process it.

Reply:
done, pushed a new commit